[dacap-clip] New port

PolarGoose opened the pull request:
<!-- If your PR fixes issues, please note that here by adding "Fixes #NNNNNN." for each fixed issue on separate lines. -->

<!-- If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/. -->

<!-- If this PR updates an existing port, please uncomment and fill out this checklist:

- [ ] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] SHA512s are updated for each updated download.
- [ ] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [ ] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [ ] Only one version is added to each modified port's versions file.

END OF PORT UPDATE CHECKLIST (delete this line) -->

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

BillyONeal commented on ports/clip/portfile.cmake:
```diff
@@ -0,0 +1,20 @@
+    -DCLIP_ENABLE_LIST_FORMATS=ON
+    -DCLIP_EXAMPLES=OFF
+    -DCLIP_TESTS=OFF
+    -DCLIP_X11_WITH_PNG=ON)
```
```suggestion
    -DCLIP_X11_WITH_PNG=ON
  MAYBE_UNUSED_VARIABLES
    CLIP_X11_WITH_PNG # only an option when UNIX AND NOT APPLE
)
```

BillyONeal commented:
I'm a bit concerned with the name "clip" due to potential confusion with https://github.com/monatis/clip.cpp . Would you find `dacap-clip` acceptable? That would match https://repology.org/project/dacap-clip .

Everything else LGTM, thanks for the new port!

dg0yt commented:
For vcpkg, the patch could be reduced to inserting `target_include_directories` after the original `GNUInstallDirs` line.  For upstreaming, the patch should be taken as is but extended to also set the `BUILD_INTERFACE`.

Is the `include/clip.h` a good location, or should it go to a subdir?

PolarGoose committed:
[dacap-clip] Add version 1.10.0

PolarGoose commented:
@BillyONeal,

> I'm a bit concerned with the name "clip" due to potential confusion with https://github.com/monatis/clip.cpp . Would you find dacap-clip acceptable? That would match https://repology.org/project/dacap-clip .

Valid point, I have changed the name.
I have also added deploying `clip.h` to the `dacap` directory.

@dg0yt,
> For vcpkg, the patch could be reduced to inserting target_include_directories after the original GNUInstallDirs line. 

Yes, indeed, thank you for the hint.

> For upstreaming, the patch should be taken as is but extended to also set the BUILD_INTERFACE.

Yes, it is a good point. I didn't include this in the patch because it is not needed for this vcpkg

> Is the include/clip.h a good location, or should it go to a subdir?

I have changed the patch to deploy to `dacap/clip.h`

dg0yt commented:
> I have changed the patch to deploy to `dacap/clip.h`

Wait, this would change the way users would need to write `#include <...>`.
I don't know the documentation/upstream intentions, but initially it might be enough to pass `-DCMAKE_INSTALL_INCLUDEDIR=include/dacap` to change location and cmake config.

PolarGoose commented:
@dg0yt,

Yes, it is true.
I have reverted this change. Let's keep it as in the original, so that users can use `#include <clip.h>` as in the documentation.

BillyONeal committed:
Make png a target rather than host dependency.

BillyONeal reviewed (approved):
Thanks for the new port!
